### ISSUE TITLE
locallab: don't reprocess when just switching current spot

### ISSUE THREAD

**Issue by heckflosse** [Beep6581/RawTherapee#5268]
When using more than one spot and switching between spots, the preview gets reprocessed. In my opinion we can skip the reprocessing. Suggested fix:

```diff
diff --git a/rtengine/refreshmap.cc b/rtengine/refreshmap.cc
index 6bdce4598..2f1de6f09 100644
--- a/rtengine/refreshmap.cc
+++ b/rtengine/refreshmap.cc
@@ -523,7 +523,7 @@ int refreshmap[rtengine::NUMOFEVENTS] = {
     0,                // unused
     LUMINANCECURVE,   // EvLocallabSpotCreated
     LUMINANCECURVE,   // EvLocallabSpotDeleted
-    LUMINANCECURVE,   // EvLocallabSpotSelected
+    M_VOID,           // EvLocallabSpotSelected
     M_VOID,           // EvLocallabSpotName
     M_VOID,           // EvLocallabSpotVisibility
     LUMINANCECURVE,   // EvLocallabSpotShape

```

**Comment by Pandagrapher** [comment]
@heckflosse 
In fact, it’s chosen otherwise mask curve backgrounds aren’t updated when switching from another spot

Pierre

**Comment by heckflosse** [comment]
@Pandagrapher Good point. Thinking...

**Comment by heckflosse** [comment]
Well, I think the mask handling needs some changes. Let's look at the exposure mask as an example.
Although the user gets the impression that the mask mode combobox is per spot (because it's in the exposure spot ui), that's not the case. The mode is always the same for all exposure spots. That means it's at the wrong place. It should be part of the global locallab settings imho. Then we could easily fire a different event when the mask mode is set to `none`. 
We should give the whole mask stuff a review...

**Comment by Pandagrapher** [comment]
The mask mode combobox is in the tool GUI because it only applies on the selected spot (but the mode is kept if selected spot is changed). I managed it like this to have the same behavior that sharpening mask toolbar button: not saving a display setting which can be exported “by mistake”. Moreover, user would want to change the mode while playing on mask parameter, so I just let it next to. If we can access the event type in `improcoordinator`, it’s easy not to refresh the image if mask mode is `none` but only mask curve background

Pierre

**Comment by heckflosse** [comment]
Mask behaviour is a bit chaotic atm.
Example: 
1) apply this pp3 to an image 
[mask.pp3.txt](https://github.com/Beep6581/RawTherapee/files/3050421/mask.pp3.txt)
looks good. Preview and Navigator show the same
![grafik](https://user-images.githubusercontent.com/13733487/55669121-ebad5780-5873-11e9-95f6-b2525d43f8bf.png)

2) set mask mode to `show mask`
looks good. Preview and Navigator show the same
![grafik](https://user-images.githubusercontent.com/13733487/55669131-097abc80-5874-11e9-8439-18855ef713ae.png)

3) set mask mode back to `none`
now navigator still shows the mask while preview (correctly) does not
![grafik](https://user-images.githubusercontent.com/13733487/55669144-23b49a80-5874-11e9-82de-006312d87601.png)

**Comment by Pandagrapher** [comment]
@heckflosse 
With commit `2080f65`, I have made some improvements to avoid reprocessing when switching current spot. Now, when switching, reprocessing is only performed when mask preview was previously activated on selected spot (to allow its deactivation). Otherwise mask curve background update still remains functional without reprocessing.

By the way I have corrected the chaotic behavior on mask preview: only detailed image shows mask preview ("dcrop" pipeline).

For the moment, modifications have been only pushed to my fork:
https://github.com/Pandagrapher/RawTherapee

Pierre
